runtime/src/main/java/org/corfudb/runtime/LogReplicationServiceClient.java

```
@@ -0,0 +1,339 @@
+     * @param destination Cluster from which the logicalGroup should be removed from further replication
+     * @return true if destination was removed
+     */
+    public boolean removeDestination(String logicalGroup, String destination) {
+        try {
+            return IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = ClientDestinationInfoKey.newBuilder()
+                            .setClientName(clientName)
+                            .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                            .setGroupName(logicalGroup)
+                            .build();
+
+                    if (txn.isExists(LR_SOURCE_METADATA_TABLE_NAME, clientInfoKey)) {
+                        DestinationInfoVal clientDestinationIds = txn.getRecord(sourceMetadataTable, clientInfoKey).getPayload();
+                        List<String> clientDestinationIdsList = new ArrayList<>(clientDestinationIds.getDestinationIdsList());
+
+                        if (clientDestinationIdsList.remove(destination)) {
+                            clientDestinationIds = clientDestinationIds.toBuilder()
+                                    .clearDestinationIds()
+                                    .addAllDestinationIds(clientDestinationIdsList)
+                                    .build();
+                            txn.putRecord(sourceMetadataTable, clientInfoKey, clientDestinationIds, null);
+                            txn.commit();
+
+                            return true;
+                        }
+                        log.debug("Destination not found");
+                        return false;
+                    }
+                    throw new Exception(String.format("Record not found for group: %s", logicalGroup));
```

You can throw an IllegalArgument exception here IMO